### PR TITLE
Improve upstream

### PR DESCRIPTION
**Pull request by publicarray**
I've squashed some bugs, changed min pass length to 8 chars, removed old code ~~and have implemented the DEFAULT_DATETIME_FORMAT option and introduced a new DEFAULT_DATEPICKER_FORMAT option.~~ this is now in #483. These settings aren't perfect but I think it it a big improvement to what is was before.

Sorry that this is a bit much but I've already squashed things from my master branch to ease the code review.

Happy to receive feedback 😄

### CODE REVIEW

File: resources/views/Shared/Layouts/Master.blade.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ class="loadModal editUserModal" href="javascript:void(0);"><span class="icon ico
     $(function () {
         $.ajaxSetup({
             headers: {
-                'X-CSRF-Token': "<?php echo csrf_token() ?>"
+                'X-CSRF-TOKEN': $('meta[name="_token"]').attr('content')
```
Comment (jeremyquinton) [inline resources/views/Shared/Layouts/Master.blade.php]: @publicarray can you explain what this change does?

File: app/Http/routes.php [hunk 2]
```diff
@@ -384,18 +384,6 @@ function ($event_id) {
...
-             */
-            Route::get('{event_id}/tickets/questions', [
-                'as'   => 'showTicketQuestions',
-                'uses' => 'EventTicketQuestionsController@showQuestions',
-            ]);
-            Route::post('{event_id}/tickets/questions/create', [
-                'as'   => 'postCreateQuestion',
-                'uses' => 'EventTicketQuestionsController@postCreateQuestion',
-            ]);
-
```
Comment (jeremyquinton) [inline app/Http/routes.php]: @publicarray why did you remove these routes

File: resources/views/Shared/Layouts/Master.blade.php (hunk 1 above)
Comment (publicarray) [reply resources/views/Shared/Layouts/Master.blade.php]: This is more of a personal thing but I was initially going to convert the php to blade `{{ }}` for constancy.  Later I looked at the [Laravel Docs](https://laravel.com/docs/5.7/csrf#csrf-x-csrf-token) and this is how they have it. I image it might be better because we remain in the JS context and don't switch to php/blade. If you like I can remove this change. I don't have a strong option on this one but I thought going with the convention might be better. (I image it is the convention when people copy from the documentation 😏)

File: app/Http/routes.php (hunk 2 above)
Comment (publicarray) [reply app/Http/routes.php]: Unfortunately [EventTicketQuestionsController doesn't exist](https://github.com/Attendize/Attendize/search?q=EventTicketQuestionsController&unscoped_q=EventTicketQuestionsController) yet and it fixes https://github.com/Attendize/Attendize/issues/327.

**Comment by jeremyquinton** [comment]
@publicarray I went through this again I can't accept all these changes. I'm going to have to create a new branch and cherry pick the best parts out. 

Are you happy to update this PR removing certain things I mark as remove?

**Comment by publicarray** [comment]
Yea sure I need to fix up the merge conflict (as expected) anyway.

**Comment by publicarray** [comment]
I think I'll mode the date time stuff to a separate PR
